packages/checkout/widgets-lib/src/views/top-up/TopUpMenuItem.tsx

```diff
@@ -25,15 +27,19 @@ export function TopUpMenuItem({
         onClick={!isDisabled ? onClick : undefined}
         sx={isDisabled ? { opacity: '0.5', cursor: 'not-allowed' } : {}}
       >
-        <MenuItem.Icon
-          icon={icon}
-          variant={iconVariant}
-        />
+        {isDualVariantIcon(icon) ? (
+          <MenuItem.Icon
+            icon={icon}
+            variant={iconVariant}
+          />
+        ) : (
+          <MenuItem.Icon icon={icon} />
+        )}
         <MenuItem.Label size="medium">
           {heading}
         </MenuItem.Label>
         <MenuItem.IntentIcon
-          icon={intentIcon}
+          icon={intentIcon as AllSingleVariantIconKeys}
```

Comment:
so it kinda depends on how many icons we want to support in here. If we want to support Dual and Single icon variants properly, then we should maybe do as above and use the typeGuard: `isDualVariantIcon()`